content/en/docs/File Format/implementationstatus.md

```diff
@@ -0,0 +1,178 @@
+| Statistics min_value, max_value           |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+| Column index                              |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+| Offset index                              |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+| Modular encryption                        |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+| Page CRC32 checksum                       |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+| Modular encryption                        |       |        |        |       |       |
++-------------------------------------------+-------+--------+--------+-------+-------+
+
+High level data API-s for parquet feature usage
+===============================================
+
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Format                                       | C++   | Python | Java   | Go    | Rust  |
+|                                              |       |        |        |       |       |
++==============================================+=======+========+========+=======+=======+
+| Hive-style partitioning                      |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Partition pruning on the partition column    |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| External column data                         |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| RowGroup Sorting column                      |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Read / Write RowGroup metadata and data (1)  |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| RowGroup pruning using statistics            |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Read / Write page metadata and data (2)      |       |        |        |       |       |
++----------------------------------------------+-------+--------+--------+-------+-------+
+| Page pruning using projection pushdown       |       |        |        |       |       |
```

Comment:
When I think "projection pushdown" I think "column selection".  Is that what is being discussed here?  I guess I wouldn't associate it with "page pruning" in my mind.

Reply:
Yes, that's column selection. I think this is the academic language also the Apache Arrow blog refers to that under the same name: https://arrow.apache.org/blog/2022/12/26/querying-parquet-with-millisecond-latency/#projection-pushdown

Reply:
Hmm, why would page pruning use "projection pushdown" but row group pruning use "statistics"? Both actually do projection pushdown, and both use statistics in that goal, am I mistaken?

Reply:
It's a bit confusing though, because it's a different kind of "page pruning" from the below (row-based pruning using statistics or bloom filters).

It's actually not so much "pruning" then "not reading all data at once, just the columns the user is interested in". I assume all reasonable Parquet implementations to do that already, does it need to be mentioned at all?

Reply:
When I read the “pruning” word in DB context I’m always assuming “skipping large chunks”, but I 100% agree that using the same term for both can be confusing